File: backend/internal/models/allocation.go
package models

type AllocationItem struct {
	ID              string  `json:"id"`
	Department      string  `json:"department"`
	ProjectName     string  `json:"projectName"`
	Owner           string  `json:"owner"`
	Provider        string  `json:"provider"`
	AllocationModel string  `json:"allocationModel"`
	Spend           float64 `json:"spend"`
	MoMChange       float64 `json:"momChange"`
	IsTagged        bool    `json:"isTagged"`
}

type AllocationDepartmentSummary struct {
	Department string  `json:"department"`
	Projects   int     `json:"projects"`
	Spend      float64 `json:"spend"`
	Tagged     int     `json:"tagged"`
}

type AllocationSummary struct {
	TotalSpend      float64                     `json:"totalSpend"`
	ComplianceRate  float64                     `json:"complianceRate"`
	TaggedCount     int                         `json:"taggedCount"`
	UntaggedCount   int                         `json:"untaggedCount"`
	AverageMoMChange float64                    `json:"averageMomChange"`
	Departments     []AllocationDepartmentSummary `json:"departments"`
}

type AllocationResponse struct {
	Status      string           `json:"status"`
	GeneratedAt string           `json:"generatedAt"`
	Summary     AllocationSummary `json:"summary"`
	Allocations []AllocationItem `json:"allocations"`
}


type ServiceBreakdownItem struct {
	ServiceName string  `json:"serviceName"`
	UsageType   string  `json:"usageType"`
	Cost        float64 `json:"cost"`
}

File: frontend/types/allocation.ts

 export interface AllocationItem {
  id: string;
  department: string;
  projectName: string;
  owner: string;
  provider: string;
  allocationModel: string;
  spend: number;
  momChange: number;
  isTagged: boolean;
}

export interface AllocationDepartmentSummary {
  department: string;
  projects: number;
  spend: number;
  tagged: number;
}

export interface AllocationSummary {
  totalSpend: number;
  complianceRate: number;
  taggedCount: number;
  untaggedCount: number;
  averageMomChange: number;
  departments: AllocationDepartmentSummary[];
}

export interface AllocationResponse {
  status: string;
  generatedAt: string;
  summary: AllocationSummary;
  allocations: AllocationItem[];
}

 export interface AllocationInsightsProps {
  selectedDept: string;
  complianceRate: number;
  taggedCount: number;
  untaggedCount: number;
  averageMomChange: number;
}
export interface AllocationTableProps {
  allocations: AllocationItem[];
  searchTerm: string;
  setSearchTerm: (value: string) => void;
}
export interface ProjectDetailModalProps {
  project: AllocationItem | null;
  onClose: () => void;
}

export interface ServiceCost {
  serviceName: string;
  usageType: string;
  cost: number;
}

 interface PieChartItem {
  name: string;
  value: number;
  fill: string;
}

export interface AllocationDonutChartProps {
  selectedDept: string;
  pieDataWithColors: PieChartItem[];
  loading: boolean;
}

export interface AllocationFilterToolbarProps {
  selectedDept: string;
  onDeptChange: (dept: string) => void;
  tagFilter: string;
  onTagFilterChange: (tag: string) => void;
  onExport: () => void;
  disableExport: boolean;
}